providers/active_directory.go

```diff
@@ -231,8 +232,9 @@ func (s *ADProvider) getUserData(username string, password string) (goth.User, e
 		if j.Name == s.config.LDAPLastNameAttribute {
 			thisUser.LastName = j.Values[0]
 		}
-		
+
 		thisUser.RawData[j.Name] = j.Values[0]
```

Comment:
maybe it's not best practice in go but shouldn't we check if thisUser.RawData == nil?

Reply:
That's what this pr solves but by initialisation rather than a check. It's not performance limited code so I don't mind the small allocation for the map even if it's unpopulated

Reply:
It's not about saving space but about best practice and avoiding runtime panic next time.